v22.1/api-support-policy.md

```diff
@@ -0,0 +1,92 @@
+| Stable   | Supported for interfacing with third-party automated tools.               | [Backward-incompatible changes](#backward-incompatible-changes) may be introduced in new major versions.<br />[Backward-compatible changes](#backward-compatible-changes) may be introduced in new minor versions. |
+| Unstable | Supported for consumption by humans. Not supported for automation.        | [Backward-incompatible changes](#backward-incompatible-changes) may be introduced in new minor and major versions.                                                                                                 |
+| Reserved | Intended for use by CockroachDB developers. Not supported for public use. | N/A                                                                                                                                                                                                                |
+
+### Backward-incompatible changes
+
+The following changes are also known as "breaking changes":
+
+- Removal or renaming of an endpoint, [built-in function](functions-and-operators.html#built-in-functions), [cluster setting](cluster-settings.html), or session variable.
+- Removal or renaming of a SQL statement or syntax.
+- Addition, removal, or renaming of a mandatory command-line flag or HTTP field.
+- Removal or renaming of an optional command-line flag or HTTP field.
+- Change in behavior of a [built-in function](functions-and-operators.html#built-in-functions) without fixing a bug or PostgreSQL incompatibility.
+- Removal or renaming of possible values in an `ENUM` session variable or [cluster setting](cluster-settings.html).
+- Change in behavior of a [structured log event](eventlog.html) type, including the [logging channel](logging-overview.html#logging-channels) it is emitted on.
+- Renaming of a [structured log event](eventlog.html) type or payload field.
+
+### Backward-compatible changes
+
+The following list is not exhaustive:
+
+- Addition of an optional command-line flag or HTTP field.
+- Change in non-interactive [`cockroach sql`](cockroach-sql.html) shell input or output.
+- Removal or change of any functionality documented as Preview or otherwise not fully supported.
+- Marking functionality as deprecated via in-line documentation, hints, or warnings without removing it altogether.
+- Addition or removal of a metric.
+- Addition of a structured log event type or payload field.
+- Addition of a new [logging channel](logging-overview.html#logging-channels).
+
+{{site.data.alerts.callout_success}}
+Backward-incompatible changes to CockroachDB, including those affecting stable APIs, are identified in the [release notes](../releases/index.html#production-releases) for major CockroachDB versions. Users are asked to [consider backward-incompatible changes before upgrading](upgrade-cockroach-version.html#review-breaking-changes) to a new CockroachDB version.
```

Comment:
I'm not sure why we say `including those affecting stable APIs` here. We're definitely going to call out breaking changes to stable APIs. The question is: Will we call out breaking changes to unstable APIs? This page suggests that the answer is "Yes." Want to confirm that.

Reply:
I think we _generally_ call out changes to unstable APIs, but:
1. we don't have a mandate to do so, so sometimes we don't and we've been ok with it
2. when we do, we only do so in release notes and nearly never in the "backward-incompatible changes" category for notes. It's usually just "xxx change".

IMHO this separation in the handling of stable features (clear and loud callouts in GA releases) and unstable features (sometimes callouts, only in general release notes) needs to be highlighted here somehow.